### PR TITLE
Fix crash when `Pipeline::executePrefix` fail and support random point for pipeline in unit test

### LINKED ISSUE #7503: Pipeline: `"Assert task_status == ExecTaskStatus::FINALIZE fail! The state of the Task must be FINALIZE before it is destructed, but it is actually INIT"`
If `PipelineExec::executePrefix` returns an error, `PipelineTask` will not be built successfully.
But the destructor of `Task` will still be called at this time. 
This is because `Task::finalize` has not been called yet, resulting in an error being thrown.

wMNTwf8k1W

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #7503 ref https://github.com/pingcap/tiflash/pull/7372

Problem Summary:

### What is changed and how it works?
The bug has been fixed in https://github.com/pingcap/tiflash/pull/7499, here, we are just refactoring the code and adding tests.
- remove useless status `FINALIZE`, and add clearer warning logs.
-  add new failpoints `random_pipeline_model_execute_prefix_failpoint`, `random_pipeline_model_execute_suffix_failpoint` and support random point for pipeline in unit test.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by yibin87 — APPROVED** [review]
LGTM

File: dbms/src/Common/FailPoint.h [hunk 1]
```diff
@@ -66,6 +68,8 @@ class FailPointHelper
      */
     static void initRandomFailPoints(Poco::Util::LayeredConfiguration & config, const LoggerPtr & log);
 
+    static void disableRandomFailPoints(Poco::Util::LayeredConfiguration & config, const LoggerPtr & log);
```
Comment (yibin87) [inline dbms/src/Common/FailPoint.h:71]: Better add some comments to describe, it is just a "clear" operation.

Comment (SeaRise) [reply dbms/src/Common/FailPoint.h:77]: ok, comment `3. Call disableFailPoint method with parsed FailPointName and ignore Rate.` added.

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**